ocaml/xapi/xapi_pif.ml

```diff
@@ -767,6 +767,19 @@ let set_property ~__context ~self ~name ~value =
         Nm.bring_pif_up ~__context pif
     ) (self :: vlan_pifs)
 
+let pif_has_clustering_enabled ~__context (self : API.ref_PIF) =
+  let network_ref = Db.PIF.get_network ~__context ~self |> Ref.string_of in
+  (Db.Cluster.get_refs_where ~__context
+     ~expr:Db_filter_types.(Eq(Literal network_ref,Field "network")))
+  |> function
+  | []   -> false
+  | a::_ -> true
+
+let set_disallow_unplug ~__context ~self ~value =
+  if pif_has_clustering_enabled ~__context self
+  then raise Api_errors.(Server_error(cannot_change_pif_properties, [Ref.string_of self]))
```

Comment:
As we Rob said, we should introduce a new API error that is specific to clustering.